dnf/repo.py

```diff
@@ -723,7 +727,8 @@ def _handle_new_remote(self, destdir, mirror_setup=True):
 
         # setup username/password if needed
         if self.username:
-            h.setopt(librepo.LRO_USERPWD, _user_pass_str(self.username, self.password))
+            h.setopt(librepo.LRO_USERPWD,
+                     _user_pass_str(self.username, self.password, quote=False))
```

Comment:
It sounds like a nice functionality, that will be never triggered, because ```quote=False```

Reply:
I made quote attribute required, so it has to be explicitly specified for each call.